CoseSignTool/CoseSign1/Extensions/CoseSign1MessageDetachedSignatureExtensions.cs

```diff
@@ -0,0 +1,134 @@
+            return false;
+        }
+
+        Match mimeMatch = HashMimeTypeExtension.Match(contentType);
+        if(!mimeMatch.Success)
+        {
+            Trace.TraceError($"{nameof(TryGetDetachedSignatureAlgorithm)} was called on a CoseSign1Message object({@this.GetHashCode()}) with the ContentType protected header being \"{contentType}\" however it did not match the regex pattern \"{HashMimeTypeExtension}\".");
+            return false;
+        }
+
+        name = new HashAlgorithmName(mimeMatch.Groups[AlgorithmGroupName].Value.ToUpperInvariant());
+        Debug.WriteLine($"{nameof(TryGetDetachedSignatureAlgorithm)} extracted hash algorithm name: {name.Name}, returning true");
+        return true;
+    }
+
+    /// <summary>
+    /// Can be used to determine if a given CoseSign1Message is encoded detached signature for an artifact.
+    /// </summary>
+    /// <param name="this">The COSE Sign1 Message to evaluate.</param>
+    /// <returns>True if the COSE Sign1 Message is a encoded detached signature; False otherwise.</returns>
+    public static bool IsDetachedSignature(this CoseSign1Message @this) => @this.TryGetDetachedSignatureAlgorithm(out _);
+
+    /// <summary>
+    /// Computes if the encoded detached signature within the COSE Sign1 Message object matches the given artifact stream.
+    /// </summary>
+    /// <param name="this">The COSE Sign1 Message to evaluate.</param>
+    /// <param name="artifactStream">The artifact stream to evaluate.</param>
+    /// <returns>True if the detached signature in the COSE Sign1 Message matches the signature of the artifact stream; False otherwise.</returns>
+    public static bool SignatureMatches(this CoseSign1Message @this, Stream artifactStream)
+    {
+        if (!@this.TryGetHashAlgorithm(out HashAlgorithm? hasher))
+        {
+            Trace.TraceError($"{nameof(SignatureMatches)} failed to extract a valid HashAlgorithm from the provided CoseSign1Message[{@this?.GetHashCode()}]");
+            return false;
+        }
+        ReadOnlyMemory<byte> artifactHash = hasher.ComputeHash(artifactStream);
+        bool equals = @this.Content.Value.Span.SequenceEqual(artifactHash.Span);
+        Debug.WriteLine($"{nameof(SignatureMatches)} compared the two hashes with lengths ({artifactHash.Length},{@this.Content.Value.Length}) for equality and returned {equals}");
+        return equals;
+    }
+
+    /// <summary>
+    /// Computes if the encoded detached signature within the COSE Sign1 Message object matches the given artifact bytes.
+    /// </summary>
+    /// <param name="this">The COSE Sign1 Message to evaluate.</param>
+    /// <param name="artifactBytes">The artifact bytes to evaluate.</param>
+    /// <returns>True if the detached signature in the COSE Sign1 Message matches the signature of the artifact bytes; False otherwise.</returns>
+    public static bool SignatureMatches(this CoseSign1Message @this, ReadOnlyMemory<byte> artifactBytes)

```

Comment:
It looks like these two methods are identical except for taking the artifact as bytes vs. stream.  My conciseness bias would force me to refactor them so as to avoid repeating code, probably by putting the code into a private method that takes both ArtifactBytes and ArtifactStream as optional parameters. But just because it's my bias doesn't mean it's a better way to do things, so please read this as a "you might consider" and not a "you should".

Reply:
fixed in next push